app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/individual/BusinessAccountingMethodController.scala

```diff
@@ -44,50 +44,50 @@ class BusinessAccountingMethodController @Inject()(businessAccountingMethod: Bus
-      postAction = routes.BusinessAccountingMethodController.submit(id, isEditMode),
+      postAction = routes.BusinessAccountingMethodController.submit(id, isEditMode, isGlobalEdit),
       isEditMode: Boolean,
-      backUrl = backUrl(id, isEditMode, businessCount)
+      backUrl = backUrl(id, isEditMode, isGlobalEdit, businessCount)
     )
 
-  def show(id: String, isEditMode: Boolean): Action[AnyContent] = Action.async { implicit request =>
+  def show(id: String, isEditMode: Boolean, isGlobalEdit: Boolean): Action[AnyContent] = Action.async { implicit request =>
     authService.authorised() {
       withIndividualReference { reference =>
         withAccountingMethod(reference) { accountingMethod =>
           withSelfEmploymentsCount(reference) { businessCount =>
-            Ok(view(businessAccountingMethodForm.fill(accountingMethod), id, businessCount, isEditMode))
+            Ok(view(businessAccountingMethodForm.fill(accountingMethod), id, businessCount, isEditMode, isGlobalEdit))
           }
         }
       }
     }
   }
 
-  def submit(id: String, isEditMode: Boolean): Action[AnyContent] = Action.async { implicit request =>
+  def submit(id: String, isEditMode: Boolean, isGlobalEdit: Boolean): Action[AnyContent] = Action.async { implicit request =>
     authService.authorised() {
       withIndividualReference { reference =>
         businessAccountingMethodForm.bindFromRequest().fold(
           formWithErrors =>
             withSelfEmploymentsCount(reference) { businessCount =>
-              BadRequest(view(formWithErrors, id, businessCount, isEditMode))
+              BadRequest(view(formWithErrors, id, businessCount, isEditMode, isGlobalEdit))
             },
           businessAccountingMethod =>
             multipleSelfEmploymentsService.saveAccountingMethod(reference, id, businessAccountingMethod) map {
-              case Right(_) => Redirect(routes.SelfEmployedCYAController.show(id, isEditMode))
+              case Right(_) => Redirect(routes.SelfEmployedCYAController.show(id, isEditMode, isGlobalEdit))
               case Left(_) => throw new InternalServerException("[BusinessAccountingMethodController][submit] - Could not save business accounting method")
             }
         )
       }
     }
   }
 
-  def backUrl(id: String, isEditMode: Boolean, selfEmploymentCount: Int): Option[String] = {
-    if (isEditMode && selfEmploymentCount > 1) {
+  def backUrl(id: String, isEditMode: Boolean, isGlobalEdit: Boolean, selfEmploymentCount: Int): Option[String] = {
+    if (isEditMode && isGlobalEdit && selfEmploymentCount > 1) {
       Some(routes.ChangeAccountingMethodController.show(id).url)
-    } else if (isEditMode) {
-      Some(routes.SelfEmployedCYAController.show(id, isEditMode = true).url)
+    } else if (isEditMode || isGlobalEdit) {
+      Some(routes.SelfEmployedCYAController.show(id, isEditMode = true, isGlobalEdit = true).url)
     } else {
       None
     }
```

Comment:
Not sure this logic is quite right, if we are going back to the change accounting method controller, we would want to do that when in edit mode OR global edit mode and self employment count > 1.
So:
```
(isEditMode || isGlobalEdit) && selfEmploymentCount > 1
```
We would also want to make sure that the isGlobalEdit flag is provided to that controller

Next if in either edit mode or global edit mode, we should pass the current value of isGlobalEdit through so that it doesn't end up in a global edit state when originally just in edit mode.

```
def backUrl(id: String, isEditMode: Boolean, isGlobalEdit: Boolean, selfEmploymentCount: Int): Option[String] = {
  if ((isEditMode || isGlobalEdit) && selfEmploymentCount > 1) {
    Some(routes.ChangeAccountingMethodController.show(id, isGlobalEdit).url)
  } else if (isEditMode || isGlobalEdit) {
    Some(routes.SelfEmployedCYAController.show(id, isEditMode = true, isGlobalEdit = isGlobalEdit).url)
  } else {
    None
  }
}
```